core/templates/components/forms/custom-forms-directives/image-uploader.directive.ts

```diff
@@ -51,18 +53,52 @@ angular.module('oppia').directive('imageUploader', [
             return 'This file is not recognized as an image.';
           }
 
-          if (!file.type.match('image/jpeg') &&
-            !file.type.match('image/gif') &&
-            !file.type.match('image/jpg') &&
-            !file.type.match('image/png') &&
-            !file.type.match('image/svg\\+xml')) {
+          var imageTypeMapping = {
+            jpeg: {
+              format: 'image/jpeg',
+              condition: [/jp(e?)g$/, /\.jp(e?)g$/],
+            },
+            jpg: {
+              format: 'image/jpg',
+              fileType: /jp(e?)g$/,
+              fileExtension: /\.jp(e?)g$/,
+            },
+            gif: {
+              format: 'image/gif',
+              fileType: /gif$/,
+              fileExtension: /\.gif$/,
+            },
+            png: {
+              format: 'image/png',
+              fileType: /png$/,
+              fileExtension: /\.png$/,
+            },
+            svg: {
+              format: 'image/svg\\+xml',
+              fileType: /svg\+xml$/,
+              fileExtension: /\.svg$/,
+            }
+          };
+
+          var imageFormatCheck = true;
```

Comment:
A "check" is a function. imageHasValidFormat, imageHasValidExtension would probably be better names that imply boolean variables.

Reply:
Done.